hk_adc/hk_adc.c

```diff
@@ -0,0 +1,504 @@
+
+    err = HK_ADC_ERR_CURR_READ_FAILED;
+    return err;
+}
+
+
+
+
+/**
+ * @brief HK ADC read CITIROC temperature (deg. C) function
+ *
+ * This function reads the raw voltage value (Vtemp) as read by the ADC and is
+ * returned in the unsigned 16-bit variable pointer passed to the function as an
+ * argument. ADC measures raw Vtemp value where Vtemp is equivalent to CITIROC
+ * ASIC temperature value. Conversion to actual temperature value is performed
+ * by the ground software (CITIROC UI tool).
+ *
+ * @param citi_temp[out]     pointer to unsigned 16-bit variable to store the result in.
+ * @return HK_ADC_ERR_TEMP_READ_FAILED if any error occurs otherwise HK_ADC_NO_ERR.
+ *
+ */
+
+int hk_adc_conv_read_citi_temp(uint16_t * citi_temp)
+{
+    uint8_t err = HK_ADC_ERR_TEMP_READ_FAILED;
+
+    uint8_t send_buffer[2] = {0};
+
+    err = hk_adc_reg_read(HK_ADC_REG_CONFIG, &read_value);
+
+    if (err == HK_ADC_NO_ERR)
+    {
+        // Let the rest of the configurations remain intact, just change MUX
+        // input config and start single-shot conversion
+        read_value = read_value & 0x8fff;
+        read_value = read_value | HK_ADC_CONFIG_MUX_SINGLE_2;
+
+        send_buffer[0] = (read_value >> 8) | HK_ADC_CONFIG_OS_SINGLE_CONV;
+        send_buffer[1] = read_value;
+
+        // Write changed MUX input back to the CONFIG register
+        err = hk_adc_reg_write(HK_ADC_REG_CONFIG, &send_buffer[0]);
+
+        if (err == HK_ADC_NO_ERR)
+        {
+            //INSERT DELAY FN (1sec), do not use WHILE Loops
+            //Wait until OS bit turns 1
+            do{
+                hk_adc_reg_read(HK_ADC_REG_CONFIG, &read_value);
+                os_bit = read_value >> 15;
+            }while(os_bit == HK_ADC_CONFIG_OS_NOT_READY);
+
+            err = hk_adc_reg_read(HK_ADC_REG_CONVERSION, &read_value);
+            if (err == HK_ADC_NO_ERR)
+            {
+                // Right shift the received result by 4 bits. Conversion value
+                // are 12 most significant bits
+                read_value = read_value >> 4;
+                *citi_temp = read_value;
+                return err;
+            }
+        }
+    }
+
+    err = HK_ADC_ERR_TEMP_READ_FAILED;
+    return err;
+}
+
+
+
+
+/**
+ * @brief HK ADC register write function
+ *
+ * This function performs register write operation. Usually, the value to be
+ * written is a 16-bit value as all the registers accessible for a write
+ * operation in HK ADC are 16-bit registers.
+ *
+ * @param reg[in]              Variable of enumeration type #hk_adc_register_t
+ *                             (defined in hk_adc.h) indicating which register
+ *                             is to be written.
+ * @param write_buffer[in]     pointer to an unsigned byte array containing data
+ *                             to be written.
+ * @return HK_ADC_ERR_WRITE_FAILED if any error occurs otherwise HK_ADC_NO_ERR.
+ *
+ */
+int hk_adc_reg_write(hk_adc_register_t reg, uint8_t *write_buffer)
```

Comment:
Missing a `static` here (for clarity, it would probably not be an issue globally).

Reply:
Added now.